### PR TITLE
Updated to included smoothing.

### PR DESCRIPTION
**Pull request by jyklly**

### CODE REVIEW

File: dzmetrics/analysis.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+"""
+this file basically runs analysis trying to somewhat mimic what would happen in datazilla.
+"""
+
+import pdb
```
Comment (k0s) [inline dzmetrics/analysis.py:5]: No need for this import

File: dzmetrics/analysis.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+from exp_smooth import exp_smooth
+from ttest import welchs_ttest_internal
+from fdr import rejector
+from numpy import mean, std
+
+def analysis(inputfn):
+    
+    """
+    note some of the following code will be redundant once datazilla is up and running
+    """
```
Comment (k0s) [inline dzmetrics/analysis.py:17]: This should probably be a comment, not a docstring.  We should actually put a docstring that tells what the function does

File: dzmetrics/exp_smooth.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+"""
+This function takes in new and old smoothed values for mean, sd and sample size and returns an exponentially smoothed mean, sd and sample size.
```
Comment (carljm) [inline dzmetrics/exp_smooth.py:2]: Docstring should be wrapped to 72 columns.

File: dzmetrics/exp_smooth.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+"""
+def exp_smooth(new_n, new_s, new_m, old_n, old_s, old_m, a = 0.05):
```
Comment (carljm) [inline dzmetrics/exp_smooth.py:4]: Need to add unit tests in the `tests/` directory for this function.

**Comment by carljm** [comment]
`analysis.py`, `dataproc.py`, and the `data` directory, if they belong in this repo at all, should be moved out of the actual `dzmetrics` library and into a new top-level `scripts/` or '`tools/` directory. They aren't really part of the dzmetrics library, they're just development tools/aids.

If there are manual sanity checks you're doing repeatedly using `analysis.py`, it'd be ideal to capture those as automated unit tests for the appropriate code in `dzmetrics`, if possible.

**Comment by jyklly** [comment]
I guess analysis.py, dataproc.py and data directory do not belong in this repo. This was just to get/store data and run a process which would be similar to what would happen in datazilla. It may be worth keeping a copy for reference but in any case it will be available in my repo. Sorry about that! As you know I am a statistician and not a dev so I apologize if some of my code is not in the right style. I'll try make the relevant changes and make another push.

**Comment by carljm** [comment]
If those files are useful to you in working on new metrics code, I've got no problem with them being in the repo, just prefer to have them segregated in a separate tools/scripts directory. And no worries or need to apologize re code style, I'm aware you're a statistician (and very grateful for the statistical help!), just making overly-brusque comments so as not to lose track of things that should be updated :-) If you have trouble with e.g. unit tests I'm happy to do them, just let me know.

**Comment by jyklly** [comment]
Cheers Carl! Will keep you posted.

**Pull request closed without merging**